flamingpy/codes/graphs/egraph.py

```diff
@@ -214,3 +217,132 @@ def draw_adj(self, **kwargs):
 
         adj = self.adj_generator(sparse=False)
         return plot_mat_heat_map(adj, **kwargs)
+
+    def add_qubit(
+        self,
+        qubit: Union[None, tuple] = None,
+        neighbors: Union[None, list] = None,
+        macro: Union[None, tuple] = None,
+    ) -> None:
+        """Add qubit to EGraph connected to neighbors.
+
+        Args:
+            qubit (tuple[int, int, int] or None): qubit to add. If a 3-tuple,
+                the qubit is added in that position. If qubit is None, the qubit is
+                positioned one unit further than the maximum position in the z
+                direction, in position (x_max + 1, 0, 0).
+
+            neighbours (list[int], list[tuple], or None): neighbors of qubit specified
+                with indices or positions.
+
+            macro (tuple[int, int, int], optional): the macronode into which to add
+                the qubit. The EGraph must be macronized.
+        """
+
+        # Makes sure that input qubit value and type is supported
+        if isinstance(qubit, tuple):
+            if len(qubit) != 3:
+                raise ValueError(
+                    "The position should be a 3-tuple, but it was given a "
+                    + f"{len(qubit)}D tuple."
+                )
+            if qubit in self:
+                raise ValueError(
+                    f"Could not add qubit because there is already a node in position {qubit}."
+                )
+        elif qubit is None:
+            x_max = max(map(lambda tup: tup[0], self.nodes()))
+            qubit = (x_max + 1, 0, 0)
+        else:
+            raise TypeError(
+                "Qubit type not supported. Excepted 3-tuple or None, but was"
+                + f" given {type(qubit)}"
+            )
+
+        # Check that if macro is not None, then self should be macronized EGraph
+        if isinstance(macro, tuple) and self.macro_to_micro is None:
+            raise ValueError(
+                "Cannot add a qubit to macronode because current EGraph is not macronized."
+            )
```

Comment:
Based on comment above:
```suggestion
        if add_to_macronode and self.macro_to_micro is None:
            raise ValueError(
                "Cannot add a qubit to macronode because current EGraph is not macronized."
            )
```